rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java

```diff
@@ -439,5 +459,168 @@ public Sw360Module() {
             abstract public String getShortname();
         }
 
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        @JsonIgnoreProperties({
+                "revision",
```

Comment:
If you want to update (put for releases/projects) a sw360 resource you have to set the revision of the object.

Reply:
@bs-ondem how have you solved that issue? How were you able to update the release without having the revision.

Reply:
The update method in the ProjectController takes the revision of the object from the database. But this can cause race condition. We should remove the revision from the blacklist and force the client to specify the revision in the request.

Reply:
Yes for Projects you have the updateProjectFromRequest method to update the revision.
But for the releases the revision is missing.

Yes the client should specify the revision for update (PUT) requests.

Reply:
maybe it is an idea to have the same implementation as done with the project update (line 133 in https://github.com/bsinno/sw360portal/blob/bs-ondem/addNewEndpoints_searchAndMore/rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java).

I am not sure if we would allow to set the revision of the REST endpoint. Also this would prevent us from using a different back end technology.  For me, the revision id is something couchdb scific I am not sure if this shall be exposed.

Rather I would prefer to have the server returning a 409 http code indicating that there is an update conflict.

Reply:
How do you want to detect an update conflict without the revision?